src/c++/library/http_client.cc

```diff
@@ -48,6 +48,7 @@ extern "C" {
 
 #ifdef _WIN32
 #define strncasecmp(x, y, z) _strnicmp(x, y, z)
+#undef min
```

Comment:
I would prefer this getting undefined by using `/DNOMINMAX` at compile time that way we won't have to worry about `undef`ing these macros in each file. If adding this compiler flag breaks the build in an unforeseen way, I'll concede to using this `undef` with a note explaining why `NOMINMAX` breaks.

Reply:
I found out yesterday that the `NOMINMAX` did not fix the build (maybe I read the wrong build output last Thursday), but the change suggested by [MS eng](https://github.com/RandySheriffH/triton_client/commit/ff6cef643eb176d41fce0c8c18fdb531e91c7816) worked, so I think it is better to stick with this change.

Reply:
Where did you put the NOMINMAX? It's harder for this file individually, since you'd have to check where the windows.h file is actually imported and then call it before you include `windows.h` in that file.

We might want to go through the repo at some point and do that define before every windows.h include, since we never use their min/max definition, as far as I know.

Reply:
I filed a ticket for the `std::min` and `std::max` issue on windows, so we can look for a more complete solution DLIS-4344.